### ISSUE TITLE
`quote-props`: check keys of concise methods

### ISSUE THREAD

**Issue by DMartens** [eslint-stylistic/eslint-stylistic#291]
### Clear and concise description of the problem

The rule `quote-props` ignores quotes around keys of concise methods.
For example this is never reported regardless of the used options:
```javascript
const object = { 'method'() {} }
```

### Suggested solution

Introduce a new option for handling concise methods, e.g. `ignoreConciseMethods`, which defaults to true to make this a non-breaking change.

### Additional context

The only mention of why I can find from the eslint team is [this PR](https://github.com/eslint/eslint/pull/2286), which just boils down to "we should ignore them by default".

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue
